src/vms/pvm/etna-builder/builder.test.ts

```diff
@@ -1007,4 +1012,103 @@ describe('./src/vms/pvm/etna-builder/builder.test.ts', () => {
     });
   });
+
+  describe('ConvertSubnetTx', () => {
+    it('should create an ConvertSubnetTx', () => {
+      const utxoInputAmt = BigInt(50 * 1e9);
+      const utxos = testUtxos();
+      const signer = new ProofOfPossession(
+        blsPublicKeyBytes(),
+        blsSignatureBytes(),
+      );
+      const pChainOwner = PChainOwner.fromNative([testAddress1], 1);
+
+      const validator = ConvertSubnetValidator.fromNative(
+        nodeId,
+        BigInt(1 * 1e9),
+        BigInt(0 * 1e9),
+        signer,
+        pChainOwner,
+        pChainOwner,
+      );
+
+      const unsignedTx = newConvertSubnetTx(
+        {
+          fromAddressesBytes,
+          feeState,
+          utxos,
+          subnetAuth: [0],
+          subnetId: Id.fromHex(testSubnetId).toString(),
+          address: testAddress1,
+          validators: [validator],
+          chainId: 'h5vH4Zz53MTN2jf72axZCfo1VbG1cMR6giR4Ra2TTpEmqxDWB',
+        },
+        testContext,
+      );
+
+      const { baseTx } = unsignedTx.getTx() as ConvertSubnetTx;
+      const { inputs, outputs } = baseTx;
+
+      const [amountConsumed, expectedAmountConsumed, expectedFee] =
+        checkFeeIsCorrect({
+          unsignedTx,
+          inputs,
+          outputs,
+          feeState,
+        });
+
+      expect(amountConsumed).toEqual(expectedAmountConsumed);
+
+      const expectedTx = new ConvertSubnetTx(
+        AvaxBaseTx.fromNative(
+          testContext.networkID,
+          testContext.pBlockchainID,
+          [getTransferableOutForTest(utxoInputAmt - expectedFee)],
+          [getTransferableInputForTest(utxoInputAmt)],
+          new Uint8Array(),
+        ),
+        Id.fromHex(testSubnetId),
+        Id.fromString('h5vH4Zz53MTN2jf72axZCfo1VbG1cMR6giR4Ra2TTpEmqxDWB'),
+        new Bytes(testAddress1),
+        [validator],
+        Input.fromNative([0]),
+      );
+      expectTxs(unsignedTx.getTx(), expectedTx);
+    });
+  });
+
+  describe.each([
+    {
+      name: 'should throw error if weight on a validator is 0',
+      validator: ConvertSubnetValidator.fromNative(
+        nodeId,
+        BigInt(0 * 1e9),
+        BigInt(0 * 1e9),
+        new ProofOfPossession(blsPublicKeyBytes(), blsSignatureBytes()),
+        PChainOwner.fromNative([testAddress1], 1),
+        PChainOwner.fromNative([testAddress1], 1),
+      ),
+    },
+  ])('$name', ({ validator }) => {
+    const utxos = testUtxos();
+    try {
+      newConvertSubnetTx(
+        {
+          fromAddressesBytes,
+          feeState,
+          utxos,
+          subnetAuth: [0],
+          subnetId: Id.fromHex(testSubnetId).toString(),
+          address: testAddress1,
+          validators: [validator],
+          chainId: 'h5vH4Zz53MTN2jf72axZCfo1VbG1cMR6giR4Ra2TTpEmqxDWB',
+        },
+        testContext,
+      );
+    } catch (error) {
+      expect((error as Error).message).toEqual(
+        'Validator weight must be greater than 0',
+      );
+    }
+  });
```

Comment:
Unnecessary describe.each block. Can we move this test to be nested in `describe('ConvertSubnetTx')`